thrift/src/main/java/com/linecorp/armeria/server/thrift/THttpHeaderInjectingService.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.thrift;
+
+import static com.linecorp.armeria.server.thrift.THttpService.HTTP_HEADERS;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.DefaultRpcResponse;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.RpcRequest;
+import com.linecorp.armeria.common.RpcResponse;
+import com.linecorp.armeria.common.util.Exceptions;
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.SimpleDecoratingService;
+
+/**
+ * A {@link SimpleDecoratingService} that injects {@link HttpHeaders} when using {@link THttpService}.
```

Comment:
A Simple ... -> `Decorates a {@link Service} to inject ...` (for consistency with other decorators)